backon/src/blocking_sleep.rs

```diff
@@ -53,3 +53,15 @@ impl BlockingSleeper for StdSleeper {
         std::thread::sleep(dur)
     }
 }
+
+/// A no_std blocking sleeper
+#[cfg(feature = "embassy-sleep")]
+#[derive(Clone, Copy, Debug, Default)]
+pub struct EmbassySleeper;
```

Comment:
If `embassy` can be used in both blocking and async contexts, I propose creating a new file named `embassy_timer_sleep.rs`. In this file, we would implement both `BlockingSleeper` and `Sleeper` for `EmbassyTimerSleeper`.